packages/Kokkos/src/DTK_PointCloud_def.hpp

```diff
@@ -105,8 +104,8 @@ class LocalBoundingBoxFunctor
     {
         for ( size_t d = 0; d < 3; ++d )
         {
-            dst[d] = std::numeric_limits<SC>::max();
-            dst[d + 3] = -std::numeric_limits<SC>::max();
+            dst[d] = KokkosHelpers::numericLimitsMax<SC>();
+            dst[d + 3] = -KokkosHelpers::numericLimitsMax<SC>();
```

Comment:
Some of the code we may potentially rely on is in KokkosKernels subpackage of Tpetra (for instance, local sparse matrices). Thus, I think relying on Kokkos code in Tpetra is fine. 

In general, I agree with @Rombur that we should use existing code rather than writing our own, as most likely the corner cases for were already identified. For instance, `Kokkos' ArithTraits` works well with `Kokkos::complex` (can't use `std::complex` anyway on the device).